src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java

```diff
@@ -0,0 +1,38 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MacroDriver implements Job2dDriver {
+    private List<DriverCommand> macro = new ArrayList<>();
+
+    @Override
+    public void operateTo(int x, int y) {
+        macro.add(new OperateToCommand(x, y));
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        macro.add((new SetPositionCommand(x, y)));
+    }
+
+    public List<DriverCommand> getMacro(){
+        return macro;
+    }
```

Comment:
Please use "deep copy" feature to return a copy of macro, so client:
-  has full control over the returned command 
- -and cannot modify the state of MacroDriver